File: src/php/t2/s1.php
<?php

declare(strict_types=1);

function number_to_ordinal(int $number): string
{
    if (0 === $number) {
        return '0';
    }

    $ending = 'th';

    if (!in_array(($number % 100), [11, 12, 13])) {
        switch ($number % 10) {
            // Handle 1st, 2nd, 3rd
            case 1:
                $ending = 'st';
                break;

            case 2:
                $ending = 'nd';
                break;

            case 3:
                $ending = 'rd';
                break;
        }
    }

    return $number.$ending;
}

File: src/go/t2/s1.go
package main

import "strconv"

func NumberToOrdinal(number int) string {
	if number == 0 {
		return "0"
	}

	special := []int{11, 12, 13}
	_, found := Find(special, number % 100)

	ending := "th"

	if !found {
		switch number % 10 {
		case 1:
			ending = "st"
		case 2:
			ending = "nd"
		case 3:
			ending = "rd"
		}
	}

	return strconv.Itoa(number) + ending
}

func Find(slice []int, val int) (int, bool) {
	for i, item := range slice {
		if item == val {
			return i, true
		}
	}
	return -1, false
}